### PR TITLE
cert-manager: optional webhook

### PR DESCRIPTION
**Pull request by jakolehm**
Webhook makes initial cluster deployment very slow.. this PR allows to disable it.

### CODE REVIEW

File: e2e/drone.sh [hunk 1]
```diff
@@ -26,7 +26,7 @@ pharos license assign --help || exit $?
 pharos license inspect --help || exit $?
 
 # Test cluster bootstrapping
-timeout 600 pharos up -y -c e2e/digitalocean/cluster.yml --tf-json e2e/digitalocean/tf.json || exit $?
+timeout 700 pharos up -y -c e2e/digitalocean/cluster.yml --tf-json e2e/digitalocean/tf.json || exit $?
```
Comment (jakolehm) [inline e2e/drone.sh]: Had to raise timeout limit, it was too close.

File: e2e/drone.sh [hunk 2]
```diff
@@ -75,8 +75,8 @@ ssh_userhost="root@${pharos_worker_host}"
 scp -o StrictHostKeyChecking=no -i "${ssh_key}" pharos-cluster*.gem "${ssh_userhost}:"
 
 ssh -o StrictHostKeyChecking=no -o ServerAliveInterval=30 -i "${ssh_key}" "${ssh_userhost}" -- "sudo bash -c '\
-  export http_proxy=http://10.133.37.156:8888 HTTP_PROXY=http://10.133.37.156:8888 https_proxy=http://10.133.37.156:8888 HTTPS_PROXY=http://10.133.37.156:8888;
-  apt-get update && apt-get -y install ruby build-essential ruby-dev && \
+  export http_proxy=http://10.133.37.156:8888 HTTP_PROXY=http://10.133.37.156:8888 https_proxy=http://10.133.37.156:8888 HTTPS_PROXY=http://10.133.37.156:8888 DEBIAN_FRONTEND=noninteractive;
```
Comment (jakolehm) [inline e2e/drone.sh]: Had to fix this because `libssl` upgrade prompted something.

File: e2e/drone_setup.sh [hunk 3]
```diff
@@ -10,15 +10,22 @@ fi
 cd e2e/digitalocean
 terraform init
 
-until terraform apply -auto-approve -var "cluster_name=${DRONE_BUILD_NUMBER}-do" -var "image=$1"
+WORKER_UP=${WORKER_UP:-false}
+if [ "${WORKER_UP}" = "true" ]; then
```
Comment (jakolehm) [inline e2e/drone_setup.sh]: Reason for this: we should only provision extra machine if we test "worker up".

File: e2e/drone_setup.sh [hunk 4]
```diff
@@ -10,15 +10,22 @@ fi
...
+until terraform apply -auto-approve -var "cluster_name=${DRONE_BUILD_NUMBER}-do" -var "image=$1" -var "worker_up_count=${WORKER_UP_COUNT}"
 do
   echo "Apply failed... trying again in 5s"
   sleep 5
 done
 
 terraform output -json > tf.json
 
-if [ "${WORKER_UP:-false}" = "up" ]; then
+if [ "${WORKER_UP}" = "true" ]; then
```
Comment (jakolehm) [inline e2e/drone_setup.sh]: We didn't really test "worker up" at all..

File: e2e/drone_setup.sh [hunk 5]
```diff
@@ -10,15 +10,22 @@ fi
 cd e2e/digitalocean
 terraform init
 
-until terraform apply -auto-approve -var "cluster_name=${DRONE_BUILD_NUMBER}-do" -var "image=$1"
+WORKER_UP=${WORKER_UP:-false}
+if [ "${WORKER_UP}" = "true" ]; then
+  WORKER_UP_COUNT="1"
```
Comment (kke) [inline e2e/drone_setup.sh]: Maybe it could just have this variable in the ENV config.

Comment (jakolehm) [reply e2e/drone_setup.sh]: Done.

File: e2e/drone_setup.sh [hunk 6]
```diff
@@ -10,15 +10,17 @@ fi
...
+until terraform apply -auto-approve -var "cluster_name=${DRONE_BUILD_NUMBER}-do" -var "image=$1" -var "worker_up_count=${WORKER_UP_COUNT}"
 do
   echo "Apply failed... trying again in 5s"
   sleep 5
 done
 
 terraform output -json > tf.json
 
-if [ "${WORKER_UP:-false}" = "up" ]; then
+if [ "${WORKER_UP_COUNT}" != "0" ]; then
```
Comment (kke) [inline e2e/drone_setup.sh]:
```suggestion
if [ "${WORKER_UP_COUNT}" -gt "0" ]; then
```

**Pull request merged**